src/modules/aws.rs

```diff
@@ -48,9 +48,11 @@ fn get_aws_region_from_config(context: &Context, aws_profile: Option<&str>) -> O
 
 fn get_aws_profile_and_region(context: &Context) -> (Option<Profile>, Option<Region>) {
     match (
-        context
-            .get_env("AWS_VAULT")
-            .or_else(|| context.get_env("AWS_PROFILE")),
+        context.get_env("AWSU_PROFILE").or_else(|| {
+            context
+                .get_env("AWS_VAULT")
+                .or_else(|| context.get_env("AWS_PROFILE"))
+        }),
```

Comment:
Rather than nesting the profile look up here maybe it would be better to iterate through a list of env vars using a find_map, something like

```rust
let profile_env_vars = vec!["AWSU_PROFILE", "AWS_VAULT", "AWS_PROFILE"];  
let profile = profile_env_vars.iter().find_map(|env_var| context.get_env(env_var));
```

Reply:
Probably!  My rust-fu is pretty much at beginner level.

Reply:
No worries we all have to start somewhere, I think writing this module might have been my first meaningful bit of rust 😀.